---
title: C28125
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C28125
helpviewer_keywords:
- C28125
ms.assetid: 3f7b8db4-6a64-4480-919e-3f3ceca7f948
author: mikeblome
ms.author: mblome
manager: markl
ms.workload:
- multiple
ms.openlocfilehash: e66e6c3eb98fdb51c6d47c70a29240ed96a8fa04
ms.sourcegitcommit: 174c992ecdc868ecbf7d3cee654bbc2855aeb67d
ms.translationtype: MT
ms.contentlocale: zh-CN
ms.lasthandoff: 12/06/2019
ms.locfileid: "74879134"
---
# <a name="c28125"></a>C28125
警告 C28125：该函数必须从 try/except 块中调用

 驱动程序正在调用一个函数，该函数必须从 try/except 块中调用，如[ProbeForRead](/windows-hardware/drivers/ddi/content/wdm/nf-wdm-probeforread)、 [ProbeForWrite](/windows-hardware/drivers/ddi/content/wdm/nf-wdm-probeforwrite)、 [MmProbeAndLockPages](/windows-hardware/drivers/ddi/content/wdm/nf-wdm-mmprobeandlockpages)。

## <a name="example"></a>示例
 下面的代码示例将生成此警告：

```cpp
ProbeForRead(addr, len, 4);
```

 下面的代码示例可避免此警告：

```cpp
__try
{
    ProbeForRead(addr, len, 4);
}
__except(EXCEPTION_EXECUTE_HANDLER)
{
    Status = GetExceptionCode();
    ... report error status
}
```
